web/src/components/datatable.tsx

```diff
@@ -0,0 +1,77 @@
+import { For, JSX, Show, createMemo, createSignal } from 'solid-js';
+
+export interface DataTableColumn<T> {
+  label: string;
+  get: (x: T) => JSX.Element;
+  sortBy?: (x: T) => string;
+}
+export interface DataTableProps<T> {
+  columns: DataTableColumn<T>[];
+  items: T[];
+}
+export function DataTable<T>(props: DataTableProps<T>): JSX.Element {
+  const [sortComparator, setSortComparator] = createSignal<
+    ((x: T, y: T) => number) | undefined
+  >();
+  const sortedItems = createMemo(() => {
+    const currentSortComparator = sortComparator();
+    if (currentSortComparator == undefined) {
+      return props.items;
+    } else {
+      const items = [...props.items];
+      items.sort(currentSortComparator);
+      return items;
+    }
+  });
+  return (
+    <>
+      <table class="font-sans border-collapse text-left w-full">
+        <thead class="text-xs uppercase text-gray-400 bg-gray-700">
+          <tr>
+            <For each={props.columns}>
+              {(col) => (
+                <th class="px-4 py-3">
+                  <Show when={col.sortBy} fallback={col.label}>
+                    <a
+                      href=""
+                      class="text-gray-400"
+                      onClick={(event) => {
+                        event.preventDefault();
+                        setSortComparator(() =>
+                          col.sortBy ? keyComparator(col.sortBy) : undefined,
+                        );
+                      }}
```

Comment:
Might be worth to move this into a function defined above the TSX.

Reply:
35f9d99e08e6cca050ebf07c5532fb7b80f1553f